File: front/src/domain/use-cases/auth/LoginUseCase.ts
import { AuthSession } from '../../entities/AuthSession';
import { IAuthRepository } from '../../repositories/IAuthRepository';

export class LoginUseCase {
  constructor(private authRepository: IAuthRepository) {}

  async execute(username: string, password: string): Promise<AuthSession> {
    if (!username || !username.trim()) {
      throw new Error('El usuario es requerido.');
    }
    if (!password) {
      throw new Error('La contraseña es requerida.');
    }
    return await this.authRepository.login(username.trim(), password);
  }
}

File: Nueva carpeta/hospital_movil/lib/features/auth/domain/usecases/login_usecase.dart
import 'package:equatable/equatable.dart';
import '../../../../core/usecase/usecase.dart';
import '../entities/auth_session.dart';
import '../repositories/auth_repository.dart';

class LoginParams extends Equatable {
  final String username;
  final String password;

  const LoginParams({
    required this.username,
    required this.password,
  });

  @override
  List<Object?> get props => [username, password];
}

class LoginUseCase implements UseCase<AuthSession, LoginParams> {
  final AuthRepository repository;

  LoginUseCase(this.repository);

  @override
  Future<Result<AuthSession>> call(LoginParams params) async {
    return await repository.login(
      username: params.username,
      password: params.password,
    );
  }
}
